app/components/round-tracker.js

```diff
@@ -0,0 +1,145 @@
+'use strict'
+import React, { useState, useEffect } from 'react'
+import { createUseStyles } from 'react-jss'
+import StatusBadge from './status-badge'
+import Theme from './theme'
+import cx from 'classnames'
+
+const RoundTracker = ({ roundsStatus = [], className, ...otherProps }) => {
+  const classes = useStyles()
+
+  const [isMobile, setIsMobile] = useState(false)
+
+  useEffect(() => {
+    const mediaQuery = window.matchMedia(`(max-width: ${Theme.condensedWidthBreakPoint})`)
+    const handleResize = () => setIsMobile(mediaQuery.matches)
+
+    // Initial check
+    handleResize()
+
+    // Add event listener
+    mediaQuery.addListener(handleResize)
+
+    // Cleanup event listener
+    return () => mediaQuery.removeListener(handleResize)
+  }, [])
+
+  const renderRounds = () => {
+    if (roundsStatus.length === 0) {
+      return <div className={classes.emptyMessage}>No rounds available</div>
+    }
+
+    let visibleRounds
+    const currentRoundIndex = roundsStatus.indexOf('inProgress')
+
+    if (isMobile) {
+      if (currentRoundIndex === 0) {
+        visibleRounds = roundsStatus.slice(0, 2) // Show the first two rounds
+      } else if (currentRoundIndex === roundsStatus.length - 1) {
+        visibleRounds = roundsStatus.slice(currentRoundIndex - 1, currentRoundIndex + 1) // Show the last two rounds
+      } else {
+        visibleRounds = roundsStatus.slice(currentRoundIndex, currentRoundIndex + 2) // Show the current and next round
+      }
+    } else {
+      if (currentRoundIndex === 0) {
+        visibleRounds = ['inProgress', 'pending', 'pending']
+      } else if (currentRoundIndex < roundsStatus.length - 1) {
+        visibleRounds = roundsStatus.slice(currentRoundIndex - 1, currentRoundIndex + 2) // Ensure only showing the last 3 rounds
+      } else {
+        visibleRounds = roundsStatus.slice(-3) // Only show the last 3 rounds
+      }
+    }
+
+    return visibleRounds.map((status, index) => (
+      <React.Fragment key={index}>
+        <div className={classes.roundContainer}>
+          <div className={classes.roundNumber}>Round {roundsStatus.length - visibleRounds.length + index + 1}</div>
+          <StatusBadge
+            name={status.charAt(0).toUpperCase() + status.slice(1)}
+            status={status.toLowerCase()}
+            className={classes.badge}
+          />
+        </div>
+        {index < visibleRounds.length - 1 && <div className={classes.dash} />}
+      </React.Fragment>
+    ))
+  }
+
+  return (
+    <div className={cx(classes.centerWrapper, className)} {...otherProps}>
+      <div className={classes.roundTrackerWrapper}>
+        <div className={classes.roundTracker}>{renderRounds()}</div>
+      </div>
+    </div>
+  )
+}
+
+const useStyles = createUseStyles(theme => ({
+  centerWrapper: {
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+    height: '100%',
+  },
+  roundTrackerWrapper: {
+    backgroundColor: '#FDFDF7',
```

Comment:
Please put all these color value into theme.js and then use the value out of theme.js